advise(react/errorboundary): add & change example case (#221)
suspensive/react: packages/react/src/ErrorBoundary.en.mdx

:::tip

## Define component as `<ErrorBoundary/>`'s fallback

### ErrorBoundaryFallbackProps

If you want to deliver a declared component as `<ErrorBoundary/>`'s fallback, you can use the `ErrorBoundaryFallbackProps` type to declare the component easily.

```tsx
import type { ErrorBoundaryFallbackProps } from '@suspensive/react'

const ErrorBoundaryFallback = ({ reset, error }: ErrorBoundaryFallbackProps) => (
  <>
    <button onClick={reset}>reset</button>
    {error.message}
  </>
)

const Example = () => (
  <ErrorBoundary fallback={ErrorBoundaryFallback}>
    <ErrorAfter4s />
  </ErrorBoundary>
)
```

:::

:::tip

## Use `<ErrorBoundary/>` fallback props without prop drilling

### useErrorBoundaryFallbackProps

> If the depth of the fallback component deepens, prop drilling will not be avoided to access `error` objects and `reset` methods.

Reviewer: ```suggestion
If component using `reset` method and `error` object is nested, prop drilling cannot be avoided.
```
